### PR TITLE
adding example for sparse BC matching

### PR DESCRIPTION
**Pull request by jgrosseo**

### CODE REVIEW

File: Analysis/Tasks/run2Matcher.cxx [hunk 1]
```diff
@@ -22,14 +22,8 @@ using namespace o2::framework::expressions;
...
   Builds<aod::Run2MatchedSparse> matched_s;
-  void init(o2::framework::InitContext&)
-  {
-  }
-};
-
-struct BCMatcher {
-  BuildsExclusive<aod::BCCollisionsExclusive> matched_e;
-  Builds<aod::BCCollisionsSparse> matched;
+  BuildsExclusive<aod::BCCollisionsExclusive> bc_e;
```
Comment (aalkin) [inline Analysis/Tasks/run2Matcher.cxx:25]: Is that a good idea to build all indices in one task? There is no optimization for building multiple index tables simultaneously and if only one of the tables is requested, there is no point in building all of them.

Comment (jgrosseo) [reply Analysis/Tasks/run2Matcher.cxx:25]: Then we have to put them in different workflows even. Before it was two devices in one workflow. So it was just consuming memory.

Comment (jgrosseo) [reply Analysis/Tasks/run2Matcher.cxx:25]: Could we do the index building similarly to the expression columns automatically and once such a table is subscribed?

Comment (aalkin) [reply Analysis/Tasks/run2Matcher.cxx:25]: Yes, it is possible to use the same trick as is done with data model extended tables, though it requires giving these tables a separate origin and hardcoding their names.

Comment (jgrosseo) [reply Analysis/Tasks/run2Matcher.cxx:25]: Very good. Yes, we can have an origin for automatic spawning. What needs to be hardcoded in the table names?

Comment (aalkin) [reply Analysis/Tasks/run2Matcher.cxx:25]: Similar to what we have in `AODReaderHelpers.cxx` lines 78-86

Comment (jgrosseo) [reply Analysis/Tasks/run2Matcher.cxx:25]: OK, so each case. If users add custom ones, they would need to produce them themselves. I think this is totally OK.

Comment (aalkin) [reply Analysis/Tasks/run2Matcher.cxx:25]: Let's merge this first and I will add `internal-dpl-index-builder` device later

**Review by iarsene — APPROVED** [review]

**Pull request merged**